crates/core/c8y_smartrest/Cargo.toml

```diff
@@ -5,11 +5,12 @@ authors = ["thin-edge.io team <info@thin-edge.io>"]
 edition = "2018"
 
 [dependencies]
+agent_interface = { path = "../agent_interface" }
```

Comment:
I really don't see why a Cumulocity specific crate has to depend on the agent interface.

* What would be the impact of removing this dependency?

Reply:
The SoftwareUpdateResponse and OperationStatus are defined in the agent_interface crate, these are used in  Cumulocity specific crate. Need some more refactoring is required to move this common code to some new crate.